tests/dragonfly/snapshot_test.py

```diff
@@ -82,3 +82,24 @@ async def test_snapshot(self, df_seeder_factory, df_server):
 
         files = [f for f in os.listdir(self.tmp_dir) if f.startswith('test-periodic')]
         assert len(files) > 0 and any(f.endswith('summary.dfs') for f in files)
+
+
+@dfly_args({**BASIC_ARGS, "dbfilename": "test-shutdown"})
+class TestDflySnapshotOnShutdown(SnapshotTestBase):
+    """Test multi file snapshot"""
+    @pytest.fixture(autouse=True)
+    def setup(self, tmp_dir: Path):
+        self.tmp_dir = tmp_dir
+
+    @pytest.mark.asyncio
+    async def test_snapshot(self, df_seeder_factory, async_client, df_server):
+        seeder = df_seeder_factory.create(port=df_server.port, **SEEDER_ARGS)
+        await seeder.run(target_deviation=0.1)
+
+        start_capture = await seeder.capture()
+
+        df_server.stop()
+        df_server.start()
+        time.sleep(2)
+
+        assert await seeder.compare(start_capture)
```

Comment:
what does this test checks?
in other words how seeder relates to the ability to save snapshots?

Reply:
IIUC, we use the seeder objects to create random data and then check again with `seeder.compare` that the same data is present in the DB after replication/snapshotting. In this case, I check that the data after doing a restart stayed the same.

Reply:
please add comments explaining each step. Also now I understand `time.sleep`  here. You will see that it will cause breakages on CI. We have better ways to wait for eventual events to finish.

Reply:
Done. Also found a loading state bug with the df format so fixed it as well :)